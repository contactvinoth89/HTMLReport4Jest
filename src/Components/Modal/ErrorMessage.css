.errorContent> span{
    display: block;
    word-wrap: break-word;
    white-space: pre-wrap;
}