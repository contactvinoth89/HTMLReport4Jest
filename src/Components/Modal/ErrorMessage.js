import React from 'react';
import './ErrorMessage.css';
import PropTypes from 'prop-types';
const Convert = require('ansi-to-html');
const convert = new Convert({
    fg: '#000',
    bg: '#fff',
    newline: false,
});
export default class ErrorMessage extends React.Component {
    escapeHtml(unsafe) {
        return unsafe
            .replace(/&/g, '&amp;')
            .replace(/</g, '&lt;')
            .replace(/>/g, '&gt;')
            .replace(/"/g, '&quot;')
            .replace(/'/g, '&#039;');
    }

    createMarkup(text) {
        let result = '';
        if (Array.isArray(text) && text.length > 0) {
            for (let i = 0; i < text.length; i++) {
                result = result.concat(
                    // eslint-disable-next-line security/detect-object-injection
                    convert.toHtml(this.escapeHtml(text[i])),
                );
            }
        } else if (!Array.isArray(text) && text && text.length > 0) {
            result = result.concat(convert.toHtml(this.escapeHtml(text)));
        }

        return { __html: result };
    }

    render() {
        return (
            <div>
                <pre
                    // eslint-disable-next-line react/no-danger
                    dangerouslySetInnerHTML={this.createMarkup(
                        this.props.messages,
                    )}
                    className="errorContent"
                />
            </div>
        );
    }

    static propTypes = {
        messages: PropTypes.any,
    };

    static defaultProps = {
        messages: '',
    };
}
