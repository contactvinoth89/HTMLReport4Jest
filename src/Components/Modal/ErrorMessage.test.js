import '@testing-library/jest-dom';
import React from 'react';
import ErrorMessage from './ErrorMessage';
import { render } from '@testing-library/react';
test('Should return empty message when null', () => {
    const { container } = render(<ErrorMessage messages={null}></ErrorMessage>);
    expect(container.firstChild.firstChild).toBeEmpty();
});
test('Should return empty message when empty', () => {
    const { container } = render(<ErrorMessage messages={[]}></ErrorMessage>);
    expect(container.firstChild.firstChild).toBeEmpty();
});

test('Should return html message', () => {
    const messages = ['\x1b[30mblack\x1b[37mwhite'];
    const { container } = render(
        <ErrorMessage messages={messages}></ErrorMessage>,
    );
    expect(container.firstChild.firstChild.firstChild.textContent).toEqual(
        'blackwhite',
    );
    expect(container).toMatchSnapshot();
});
