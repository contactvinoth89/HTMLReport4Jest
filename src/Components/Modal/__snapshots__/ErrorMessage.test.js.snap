// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should return html message 1`] = `
<div>
  <div>
    <pre
      class="errorContent"
    >
      <span
        style="color:#000"
      >
        black
        <span
          style="color:#AAA"
        >
          white
        </span>
      </span>
    </pre>
  </div>
</div>
`;
